crates/cairo-coverage/src/data_loader/loaded_data.rs

```diff
@@ -0,0 +1,60 @@
+use anyhow::Context;
+use anyhow::Result;
+use cairo_lang_sierra::program::{ProgramArtifact, VersionedProgram};
+use camino::Utf8PathBuf;
+use derived_deref::Deref;
+use serde::de::DeserializeOwned;
+use std::collections::HashMap;
+use std::fs;
+use trace_data::CallTrace;
+
+type SourceSierraPath = String;
+
+#[derive(Deref)]
+pub struct LoadedDataMap(HashMap<SourceSierraPath, LoadedData>);
+
+pub struct LoadedData {
+    pub artifact: ProgramArtifact,
+    pub call_traces: Vec<CallTrace>,
+}
+
+impl LoadedDataMap {
+    pub fn load(call_trace_paths: &Vec<Utf8PathBuf>) -> Result<Self> {
+        let mut map: HashMap<SourceSierraPath, LoadedData> = HashMap::new();
+        for call_trace_path in call_trace_paths {
+            let call_trace: CallTrace = read_and_deserialize(call_trace_path)?;
+
+            let source_sierra_path = &call_trace
+                .cairo_execution_info
+                .as_ref()
+                .context("Missing key 'cairo_execution_info' in call trace")?
```

Comment:
This might be an incompatible format, right? And it's cause might be incompatible version of trace or cairo-coverage?

Reply:
Yes

Reply:
Perhaps we can suggest that user upgrades?